pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -615,6 +617,27 @@ func getSubnetIDForLB(compute *gophercloud.ServiceClient, node v1.Node) (string,
 	return "", ErrNotFound
 }
 
+// getNodeSecurityGroupIDForLB lists node-security-groups for specific nodes
+func getNodeSecurityGroupIDForLB(compute *gophercloud.ServiceClient, nodes []*v1.Node) ([]string, error) {
+	nodeSecurityGroupIDs := sets.NewString()
+
+	for _, node := range nodes {
+		nodeName := types.NodeName(node.Name)
+		srv, err := getServerByName(compute, nodeName)
+		if err != nil {
+			return nodeSecurityGroupIDs.List(), err
+		}
+
+		for _, securityGroup := range srv.SecurityGroups {
+			for _, securityGroupName := range securityGroup {
```

Comment:
:thinking: So remind me - when will a node have multiple securityGroups, and why is it appropriate to add LB rules into all of them (rather than somehow finding the "right" one)?